flex/engines/graph_db/app/cypher_read_app.cc

```diff
@@ -24,31 +24,31 @@ bool CypherReadApp::Query(const GraphDBSession& graph, Decoder& input,
 
     gs::runtime::GraphReadInterface gri(txn);
 
-    gs::runtime::Context ctx;
+    gs::runtime::Context ctx = gs::runtime::Context::InitContext();
     gs::Status status = gs::Status::OK();
     {
       ctx = bl::try_handle_all(
-          [this, &gri, &plan]() -> bl::result<runtime::Context> {
+          [this, &gri, &plan, &ctx]() -> bl::result<runtime::Context> {
             return runtime::PlanParser::get()
                 .parse_read_pipeline(gri.schema(), gs::runtime::ContextMeta(),
                                      plan)
                 .value()
-                .Execute(gri, runtime::Context(), {}, timer_);
+                .Execute(gri, std::move(ctx), {}, timer_);
           },
-          [&status](const gs::Status& err) {
+          [&status, &ctx](const gs::Status& err) {
```

Comment:
Is the capture`&ctx` safe? Since ctx has already been applied with `std::move` in `.Execute`.